authority/provisioner/k8sSA.go

```diff
@@ -0,0 +1,245 @@
+package provisioner
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+
+	"github.com/pkg/errors"
+	"github.com/smallstep/cli/crypto/pemutil"
+	"github.com/smallstep/cli/jose"
+	"golang.org/x/crypto/ed25519"
+)
+
+var (
+	// K8sSAName is the default name used for kubernetes service account tokens,
+	// because there can only be one per instance of step-ca.
+	K8sSAName   = "k8sSA-default"
+	k8sSAIssuer = "kubernetes/serviceaccount"
+)
+
+// jwtPayload extends jwt.Claims with step attributes.
+type k8sSAPayload struct {
+	jose.Claims
+	Namespace          string `json:"kubernetes.io/serviceaccount/namespace,omitempty"`
+	SecretName         string `json:"kubernetes.io/serviceaccount/secret.name,omitempty"`
+	ServiceAccountName string `json:"kubernetes.io/serviceaccount/service-account.name,omitempty"`
+	ServiceAccountUID  string `json:"kubernetes.io/serviceaccount/service-account.uid,omitempty"`
+}
+
+// K8sSA represents a Kubernetes ServiceAccount provisioner; an
+// entity trusted to make signature requests.
+type K8sSA struct {
+	Type      string  `json:"type"`
+	Name      string  `json:"name"`
+	Claims    *Claims `json:"claims,omitempty"`
+	PubKeys   []byte  `json:"pubKeys,omitempty"`
+	claimer   *Claimer
+	audiences Audiences
+	//kauthn    kauthn.AuthenticationV1Interface
+	pubKeys []interface{}
+}
```

Comment:
Does it make sense to optionally restrict the use of the provisioner to specific service accounts?

Reply:
Might be something for @mmalone  to weigh in on.

Reply:
Sure, if it's _really_ easy. Otherwise open a ticket for it and let's do it later when we've got a little less schedule pressure.

Reply:
I'll open a ticket.  We're gonna need to revisit to implement the TokenReview API anyway.

Reply:
Sounds good.